lib/Analysis/Utility.cpp

```diff
@@ -168,6 +168,8 @@ unsigned ReduceOpHelper::getThreadsReductionAxis() {
 
 bool ReduceOpHelper::isWarpSynchronous() {
   auto srcLayout = getSrcLayout();
+  if (!srcLayout)
```

Comment:
why would there be no layout on the source? This hekper is meant to be used for lowering of trgir to llvm so the tensor should have a layout

Reply:
`encoding` is not necessary for `RankedTensorType` in MLIR. For reducing `vector` to `scalar` cases, I meet these cases without layout. Do you mean every tensor in ttgir must have a layout?

Reply:
yes in ttgir every tensor must have a layout